packages/@aws-cdk/aws-sns/lib/topic-ref.ts

```diff
@@ -169,14 +169,18 @@ export abstract class TopicBase extends cdk.Construct implements ITopic {
    * @param queue The target queue
    */
   public subscribeQueue(queue: sqs.IQueue): Subscription {
-    const subscriptionName = queue.node.id + 'Subscription';
-    if (this.node.tryFindChild(subscriptionName)) {
```

Comment:
We should make IQueue extend IConstruct

Reply:
It does, but `scope: Construct` and not `scope: IConstruct`.

Reply:
Damn, right. I feel this is going to bite us at some point. But I really don't want people to need to specify `scope: IConstruct` in the initializer...